main.cpp

```diff
@@ -119,26 +131,34 @@ int main()
 	// combo system
 	// combo names displayed when letter is correctly guessed
 	std::stringstream success_combo_names("NICE! ALRIGHT! YESS! AMAZING!");
-	const std::vector<std::string> success_combo = initialize_combolist(success_combo_names);
+	const std::vector<std::string> success_combo{initialize_combolist(success_combo_names)};
 
 	// combo names displayed when letter is not correctly guessed
 	std::stringstream failed_combo_names("ARG! DAMMIT! DARN! NOO!");
-	const std::vector<std::string> failed_combos = initialize_combolist(failed_combo_names);
+	const std::vector<std::string> failed_combos{initialize_combolist(failed_combo_names)};
 
 	std::vector<std::string> wordlist;
-	initialize_wordlist("words.txt", wordlist);
+
+	if (!wordlist_path.empty())
```

Comment:
This can be done right after filling the variable, instead of after all the code in between. Think about what error message you would like to read when it doesn't work for you; right now it will say, "Unable to locate wordlist", but a month later you won't remember what "wordlist" means, you'll have to open the source code to see what's going on, etc. Better to say something like, "FATAL_ERROR: Could not find required data! (Searched upwards from `/path/to/exe`)! Missing:\n\t1. `data/words.txt` " so you'll know exactly how to fix it when you encounter the error.

Reply:
Via `assert()` correct?

Reply:
Not exactly, you want to print this to stdout / stderr and exit gracefully with a non zero exit code; `assert()` will call `abort()` instead. This isn't a situation where the program can't continue, it's a user error.